keras_cv/models/segmentation/fcn.py

```diff
@@ -0,0 +1,677 @@
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing import Rescaling
+from keras_cv.models import VGG16
+from keras_cv.models import VGG19
+
+BACKBONE_CONFIG = {
+    "vgg16": {
+        "BLOCK3": "block3_pool",
+        "BLOCK4": "block4_pool",
+        "BLOCK5": "block5_pool",
+        "DENSE_START": "fc1",
+        "DENSE_END": "predictions",
+    },
+    "vgg19": {
+        "BLOCK3": "block3_pool",
+        "BLOCK4": "block4_pool",
+        "BLOCK5": "block5_pool",
+        "DENSE_START": "fc1",
+        "DENSE_END": "predictions",
+    },
+}
+BACKBONE = {"vgg16": VGG16, "vgg19": VGG19}
+
+
+def getPoolLayers(model, pool_id):
+    """Function to get the all layers upto a certain named layer. Used only for extracting MaxPooling2D-based layers.
+
+    Args:
+        model: `tf.keras.layers.Model` or `tf.keras.Sequential`, The original model from which the layers are extracted
+        pool_id: str, The name of the layer that acts as the end layer limit for the extraction
+    Returns:
+        layers: `tf.keras.Sequential` representing the extracted layers
+    """
+    layer_names = [layer.name for layer in model.layers]
```

Comment:
It should, actually. Thank you, this does help!

Reply:
Glad it helped! I pulled it from the top of my mind, so it's worth testing whether it works

Reply:
I will test it in a while once I am done incorporating all the changes, yes

Reply:
Just a slight change

```python
tf.keras.Model(
        inputs=model.layers[0].input, outputs=backbone.get_layer(pool_id).output
 )

Should do the trick
```

Reply:
So this does not work, because using this method adds a 5th Batch dimension because the first layer input already has a batch dim, creating a 2nd batch dimension. So will have to go back to the old get_pool_layers method itself